lib/bolt/pal.rb

```diff
@@ -479,16 +347,20 @@ def generate_types
     end
 
     def run_task(task_name, targets, params, executor, inventory, description = nil)
-      in_task_compiler(executor, inventory) do |compiler|
-        params = params.merge('_bolt_api_call' => true, '_catch_errors' => true)
-        compiler.call_function('run_task', task_name, targets, description, params)
+      in_bolt_compiler do |compiler|
+        with_bolt_executor(executor, inventory) do
+          params = params.merge('_bolt_api_call' => true, '_catch_errors' => true)
+          compiler.call_function('run_task', task_name, targets, description, params)
+        end
       end
     end
 
-    def run_plan(plan_name, params, executor = nil, inventory = nil, pdb_client = nil, applicator = nil)
-      in_plan_compiler(executor, inventory, pdb_client, applicator) do |compiler|
-        r = compiler.call_function('run_plan', plan_name, params.merge('_bolt_api_call' => true))
-        Bolt::PlanResult.from_pcore(r, 'success')
+    def run_plan(plan_name, params, executor = nil, inventory = nil, pdb_client = nil)
```

Comment:
Removing the ability to pass applicator here will force PE to re-implement this function. We pass the PE version of the applicator to this function, where the PE version implements to correct functionality to pass the apply through Orch.

Refactoring the PAL init method to take applicator would be a way to solve this.

Reply:
Ahh that makes sense. I couldn't find anywhere in Bolt where we passed an applicator through, so I removed it. I do think making the Applicator an argument to either PAL init or the `with_bolt_executor` method itself would probably make the most sense. That would remove _most_ of the parameters to the PAL constructor, which are primarily there just to be later passed to the Applicator.

Reply:
Here's the function in PE where we actually create our PAL and call run_plan: https://github.com/puppetlabs/orchestrator/blob/master/ruby/bolt-plan-executor-lib/bolt_plan_executor/jruby_plan_executor.rb#L44

Hopefully that provides a little context on our use-case. From PE's perspective I don't think it matters too much specifically where the applicator can be passed, as long as 1. we can pass it to a PAL function that's a public part of the PAL class and 2. it's easy to explicitly manage the full modulepath